Source/Core/Core/IOS/WFS/WFSI.cpp

```diff
@@ -257,6 +267,26 @@ IPCCommandResult WFSI::IOCtl(const IOCtlRequest& request)
   case IOCTL_WFSI_FINALIZE_TITLE_INSTALL:
   {
     std::string tmd_path;
+    if (m_patch_type == PATCH_TYPE_2)
+    {
+      // Delete content's default.dol
+      const std::string title_content_dir =
+          StringFromFormat("/vol/%s/title/%s/%s/content/", m_device_name.c_str(),
+                           m_current_group_id_str.c_str(), m_current_title_id_str.c_str());
+      File::Delete(WFS::NativePath(title_content_dir + "default.dol"));
+
+      // Copy content's _default.dol to patch's directory
+      const std::string patch_dir =
+          StringFromFormat("/vol/%s/title/%s/%s/_patch/", m_device_name.c_str(),
+                           m_current_group_id_str.c_str(), m_current_title_id_str.c_str());
+      const std::string patch_content_dir = patch_dir + "content/";
+      File::CreateDir(WFS::NativePath(patch_dir));
+      File::CreateDir(WFS::NativePath(patch_content_dir));
+      File::Rename(WFS::NativePath(title_content_dir + "_default.dol"),
+                   WFS::NativePath(patch_content_dir + "default.dol"));
+
+      FinalizePatchInstall();
```

Comment:
PATCH_TYPE_2 doesn't get finalized through an ioctl like PATCH_TYPE_1?

Reply:
I can't find any log of `IOCTL_WFSI_FINALIZE_PATCH_INSTALL` for titles other than S4MJGD (DQXv1). It seems DQXv2 (S4SJGD) and above don't rely on that IOCtl.

Reply:
Yeah it turns out WFSISrvMoveContentsForPatch (which is normally called from IOCTL_WFSI_FINALIZE_PATCH_INSTALL) automatically gets called from ioctl 6 for PATCH_TYPE_2.